### PR TITLE
Fix issues in TaskCreateComponent

### LINKED ISSUE #329: Task definition preview doesn't work
As a user, I'm creating a new task from the app list and the (resulting definition) preview section is blank.

image

### PR DESCRIPTION
**Pull request by jvalkeal**
- Generic polish of both component ts and html.
- Add definition name validation and reflect that
  to the UI.
- Fix definition preview so that its shown immediately.
- Add logical tests to verify functionality.
- Tweak TasksService mock to return empty response when
  task is created.
- Fixes #329

### CODE REVIEW

File: ui/src/app/tasks/task-create/task-create.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        { provide: ActivatedRoute, useValue: activeRoute },
+        { provide: ToastyService, useValue: toastyService }
+      ]
+    })
+      .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(TaskCreateComponent);
+    component = fixture.componentInstance;
```
Comment (cppwfs) [inline ui/src/app/tasks/task-create/task-create.component.spec.ts]: probably want to add `toastyService.clearAll();` to the beforeEach for future tests.

File: ui/src/app/tasks/task-create/task-create.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    el = de.nativeElement;
+    fixture.detectChanges();
+    expect(de.nativeElement.disabled).toBe(true);
+
+    updateDefinitionName('validtaskname');
+    fixture.detectChanges();
+    expect(de.nativeElement.disabled).toBe(false);
+  });
+
+  it('should navigate to tasks definitions with submit', () => {
```
Comment (cppwfs) [inline ui/src/app/tasks/task-create/task-create.component.spec.ts]: want to include test for back button click as well.

File: ui/src/app/tasks/task-create/task-create.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+  it('should warn about same definition name', () => {
+    activeRoute.testParams = commonTestParams;
+    tasksService.testAppInfos = commonTestAppInfos;
+    de = fixture.debugElement.query(By.css('[id=commonDefinitionParameters] > div'));
+    el = de.nativeElement;
+
+    updateDefinitionName('validtaskname');
+    fixture.detectChanges();
+    expect(el.classList.contains('has-warning')).toBe(false);
```
Comment (cppwfs) [inline ui/src/app/tasks/task-create/task-create.component.spec.ts]: need to check that message is not show when valid.   The message is still present when definition name is valid, as discussed here:  https://github.com/spring-cloud/spring-cloud-dataflow-ui/issues/342.

Comment (cppwfs) [inline ui/src/app/tasks/task-create/task-create.component.spec.ts]: we can add this test after #342 is fixed.

**Comment by jvalkeal** [comment]
@cppwfs I fixed #342 in this PR as a separate commit. Feel free to pick or not pick it to final merge.

**Comment by cppwfs** [comment]
LGTM 
* Will add some polishing on punctuation
* thanks for fixing https://github.com/spring-cloud/spring-cloud-dataflow-ui/issues/342 in this PR

**Comment by cppwfs** [comment]
Polished, Squashed Merged.

**Pull request closed without merging**